molecule/api.py

```diff
@@ -1,26 +1,50 @@
-import pkg_resources
+import pluggy
+from importlib import import_module
 from molecule import logger
-
+import traceback
 
 LOG = logger.get_logger(__name__)
-
 try:
     from functools import lru_cache
 except ImportError:
     from backports.functools_lru_cache import lru_cache
 
 
-@lru_cache()
-def molecule_drivers(as_dict=False):
+CORE_DRIVERS = [
+    "delegated",
+    "digitalocean",
+    "docker",
+    "ec2",
+    "gce",
+    "hetznercloud",
+    "linode",
+    "lxc",
+    "lxd",
+    "openstack",
+    "podman",
+    "vagrant",
+]
 
+
+@lru_cache()
+def molecule_drivers(as_dict=False, config=None):
     plugins = {}
-    for entry_point in pkg_resources.iter_entry_points('molecule_driver'):
+    pm = pluggy.PluginManager("molecule_driver")
+    for driver in CORE_DRIVERS:
+        pm.register(import_module("molecule.driver.%s" % driver), name=driver)
+    try:
+        pm.load_setuptools_entrypoints("molecule_driver")
+    except Exception:
```

Comment:
can we better scope this exception handling?

Reply:
To clarify, you mean, use a more explicit exception type than `Exception` or?

Reply:
yes, something more explicit. I'm no fan of `except Exception`, unless its **REALLY** needed.

Reply:
I feel like it makes sense here, even if we added specific Exception catches we'd probably want the generic one at the end, I can't think of any error that we'd want to let bubble out to the user without wrapping it in the below log message (also as pointed out in the comment exceptions here shouldn't result in a crash, since the driver may not end up being used at all).